MekHQ/src/mekhq/campaign/parts/equipment/AmmoBin.java

```diff
@@ -686,11 +687,21 @@ public boolean isCompatibleAmmo(AmmoType a1, AmmoType a2) {
     public int getAmountAvailable() {
         final AmmoType thisType = (AmmoType)getType();
         if (campaign.getCampaignOptions().useAmmoByType()) {
-            AmmoStorage a = (AmmoStorage)campaign.findSparePart(part -> {
-                return part instanceof AmmoStorage
-                    && thisType.equals(((AmmoStorage)part).getType())
-                    && thisType.getMunitionType() == ((AmmoType)((AmmoStorage)part).getType()).getMunitionType();
-            });
+            Predicate<Part> predicate;
+            if (AmmoBin.ALLOWED_BY_TYPE.contains(thisType.getAmmoType())) {
+                predicate = part -> {
+                    return part instanceof AmmoStorage
+                            && thisType.equalsAmmoTypeOnly(((AmmoStorage)part).getType())
+                            && thisType.getMunitionType() == ((AmmoType)((AmmoStorage)part).getType()).getMunitionType();
+                };
+            } else {
+                predicate = part -> {
+                    return part instanceof AmmoStorage
+                            && thisType.equals(((AmmoStorage)part).getType())
+                            && thisType.getMunitionType() == ((AmmoType)((AmmoStorage)part).getType()).getMunitionType();
+                };
+            }
+            AmmoStorage a = (AmmoStorage)campaign.findSparePart(predicate);
```

Comment:
This could be done through a ternary instead to cut down on near identical code (image is because I wasn't sure if it would work or not)
![image](https://user-images.githubusercontent.com/39067288/81303978-6da91a80-904a-11ea-9000-aeeac94afcb8.png)

Reply:
In my personal opinion I think in this case the readability of the if statement is a greater benefit. The compiler is not going to care and there will be no performance loss. But if you really prefer the ternary I can put it in and test/commit it.

Reply:
If you go with the suggested option, you'll need to put in a comment explaining what to do - I personally find complicated ternary statements difficult to parse, and someone looking at that code a year from now will need it explained.